consensus/polybft/bridge_event_manager.go

```diff
@@ -0,0 +1,569 @@
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/contractsapi"
+	polybftProto "github.com/0xPolygon/polygon-edge/consensus/polybft/proto"
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/signer"
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/validator"
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/wallet"
+	"github.com/0xPolygon/polygon-edge/contracts"
+	"github.com/0xPolygon/polygon-edge/types"
+)
+
+type Runtime interface {
+	IsActiveValidator() bool
+}
+
+// BridgeEventManager is an interface that defines functions bridge message event workflow
+type BridgeEventManager interface {
+	EventSubscriber
+	Init() error
+	AddLog(chainID *big.Int, eventLog *ethgo.Log) error
+	BridgeBatch(blockNumber uint64) (*BridgeBatchSigned, error)
+	PostBlock(req *PostBlockRequest) error
+	PostEpoch(req *PostEpochRequest) error
+}
+
+var _ BridgeEventManager = (*dummyStateSyncManager)(nil)
+
+// dummyStateSyncManager is used when bridge is not enabled
+type dummyStateSyncManager struct{}
+
+func (d *dummyStateSyncManager) Init() error                                        { return nil }
+func (d *dummyStateSyncManager) AddLog(chainID *big.Int, eventLog *ethgo.Log) error { return nil }
+func (d *dummyStateSyncManager) BridgeBatch(blockNumber uint64) (*BridgeBatchSigned, error) {
+	return nil, nil
+}
+func (d *dummyStateSyncManager) PostBlock(req *PostBlockRequest) error { return nil }
+func (d *dummyStateSyncManager) PostEpoch(req *PostEpochRequest) error {
+	return nil
+}
+
+// EventSubscriber implementation
+func (d *dummyStateSyncManager) GetLogFilters() map[types.Address][]types.Hash {
+	return make(map[types.Address][]types.Hash)
+}
+func (d *dummyStateSyncManager) ProcessLog(header *types.Header,
+	log *ethgo.Log, dbTx *bolt.Tx) error {
+	return nil
+}
+
+// bridgeEventManagerConfig holds the configuration data of bridge event manager
+type bridgeEventManagerConfig struct {
+	dataDir           string
+	topic             topic
+	key               *wallet.Key
+	maxNumberOfEvents uint64
+}
+
+var _ BridgeEventManager = (*bridgeEventManager)(nil)
+
+// bridgeEventManager is a struct that manages the workflow of
+// saving and querying bridge message events, and creating, and submitting new batches
+type bridgeEventManager struct {
+	logger hclog.Logger
+	state  *State
+
+	config *bridgeEventManagerConfig
+
+	// per epoch fields
+	lock                   sync.RWMutex
+	pendingBridgeBatches   []*PendingBridgeBatch
+	validatorSet           validator.ValidatorSet
+	epoch                  uint64
+	nextBridgeEventIDIndex uint64
+	sourceChainID          uint64
+
+	runtime Runtime
+}
+
+// topic is an interface for p2p message gossiping
+type topic interface {
+	Publish(obj proto.Message) error
+	Subscribe(handler func(obj interface{}, from peer.ID)) error
+}
+
+// newBridgeEventManager creates a new instance of bridge event manager
+func newBridgeEventManager(logger hclog.Logger, state *State, config *bridgeEventManagerConfig,
```

Comment:
General comment. Wherever you log something in this file, please include the `sourceChainID` from the struct.